pkg/openthread/contrib/openthread.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2016 José Ignacio Alamos
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @author  José Ignacio Alamos <jialamos@uc.cl>
+ */
+
+#include <assert.h>
+
+#include "mutex.h"
+#include "openthread/platform/alarm.h"
+#include "openthread/platform/uart.h"
+#include "ot.h"
+#include "random.h"
+#include "thread.h"
+#include "xtimer.h"
+
+
+#ifdef MODULE_AT86RF2XX
+#include "at86rf2xx.h"
+#include "at86rf2xx_params.h"
+#endif
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifdef MODULE_AT86RF2XX     /* is mutual exclusive with above ifdef */
+#define OPENTHREAD_NETIF_NUMOF        (sizeof(at86rf2xx_params) / sizeof(at86rf2xx_params[0]))
+#endif
+
+#ifdef MODULE_AT86RF2XX
+static at86rf2xx_t at86rf2xx_dev;
+#endif
+
+#define OPENTHREAD_NETDEV2_BUFLEN (ETHERNET_MAX_LEN)
+
+static uint8_t rx_buf[OPENTHREAD_NETDEV2_BUFLEN];
+static uint8_t tx_buf[OPENTHREAD_NETDEV2_BUFLEN];
+static char ot_thread_stack[2 * THREAD_STACKSIZE_MAIN];
+static mutex_t mtx = MUTEX_INIT;
+
+void otTaskletsSignalPending(otInstance *aInstance)
+{
+        (void)aInstance;
+}
+
+#if defined(MODULE_OPENTHREAD_CLI) || defined(MODULE_OPENTHREAD_NCP)
+/* init and run OpeanThread's UART simulation (sdtio) */
+void openthread_uart_run(void)
+{
+    char buf[256];
+    uint8_t index = 0;
+    char c;
+    msg_t msg;
+
+    msg.type = OPENTHREAD_SERIAL_MSG_TYPE_EVENT;
+    msg.content.ptr = buf;
+
+    while (1) {
+        c = getchar();
+        buf[index++] = c;
+        if(c == 0x0a)
+        {
+            buf[index] = 0;
+            index = 0;
+            msg_send(&msg, openthread_get_pid());
+        }
+    }
+}
+#endif
+
+/* call this when calling an OpenThread function */
+void begin_mutex(void)
```

Comment:
hm. Does anyone see an advantage of these functions over just using the mutex functions directly?
If yes, at least they have to be static inlines.

I'd prefer direct calls, if only to make static tests easier.